### ISSUE TITLE
repo accessible from one machine, but gives Input/output error from other machine

### ISSUE THREAD

**Issue by acodispo** [borgbackup/borg#6262]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE

#### Your borg version (borg -V).

Both machines are `borg 1.1.16`

#### Operating system (distribution) and version.

Both machines are Debian GNU/Linux 11 (bullseye)

#### Hardware / network configuration, and filesystems used.

The two clients are HP laptops. The repository is on a 5TB external USB harddrive (Western Digital).

#### How much data is handled by borg?

~230 GB.

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

In both cases I'm running:

```
borg info /path/to/repo
```
On Machine 1 this results in the error below. Unmount the drive and mount on Machine 2, and the command runs fine.

## Describe the problem you're observing.

Machine 1 is my work laptop and it's where the files I'm backing up in this repo are. Backups were working fine until a few days ago. I use Borgmatic.

I tried to create an archive (using Borgmatic) and that failed, but without much helpful output, so I used borg directly. Just `borg info`.

This is the output on Machine 1:

```
$ borg info /media/acodispo/backup-linux/borg-repo
Enter passphrase for key /media/acodispo/backup-linux/borg-repo: 
Killed stale lock machine1.home@245556284038697.27844-0.
Removed stale exclusive roster lock for host machine1.home@245556284038697 pid 27844 thread 0.
Removed stale exclusive roster lock for host machine1.home@245556284038697 pid 27844 thread 0.
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 4690, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 4622, in run
    return set_ec(func(args))
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 170, in wrapper
    with Cache(repository, kwargs['key'], kwargs['manifest'],
  File "/usr/lib/python3/dist-packages/borg/cache.py", line 387, in __new__
    return local()
  File "/usr/lib/python3/dist-packages/borg/cache.py", line 379, in local
    return LocalCache(repository=repository, key=key, manifest=manifest, path=path, sync=sync,
  File "/usr/lib/python3/dist-packages/borg/cache.py", line 464, in __init__
    self.open()
  File "/usr/lib/python3/dist-packages/borg/cache.py", line 512, in open
    self.rollback()
  File "/usr/lib/python3/dist-packages/borg/cache.py", line 629, in rollback
    self._do_open()
  File "/usr/lib/python3/dist-packages/borg/cache.py", line 502, in _do_open
    self.chunks = ChunkIndex.read(fd)
  File "src/borg/hashindex.pyx", line 113, in borg.hashindex.IndexBase.read
  File "src/borg/hashindex.pyx", line 100, in borg.hashindex.IndexBase.__cinit__
  File "/usr/lib/python3/dist-packages/borg/crypto/file_integrity.py", line 32, in read
    return self.fd.read(n)
  File "/usr/lib/python3/dist-packages/borg/crypto/file_integrity.py", line 82, in read
    data = super().read(n)
  File "/usr/lib/python3/dist-packages/borg/crypto/file_integrity.py", line 32, in read
    return self.fd.read(n)
OSError: [Errno 5] Input/output error

Platform: Linux machine1 5.10.0-11-amd64 #1 SMP Debian 5.10.92-1 (2022-01-18) x86_64
Linux: Unknown Linux  
Borg: 1.1.16  Python: CPython 3.9.2 msgpack: 0.5.6.+borg1
PID: 30281  CWD: /media/acodispo
sys.argv: ['/usr/bin/borg', 'info', '/media/acodispo/backup-linux/borg-repo']
SSH_ORIGINAL_COMMAND: None
```

I tried a variety of checks on the USB harddrive, assuming it was hardware failure, but the checks seemed alright as far as I could tell.

Eventually I unmounted the drive from Machine 1 and mounted it on Machine 2, another laptop. On Machine 2 `borg info` runs fine and shows expected information.

Finally, with the harddrive still mounted on Machine 2, I tried `borg info` via SSH from Machine 1 to Machine 2. The error was the same as above with the addition of "The repository was previously located at ..." message which I expected.

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

The error is entirely reproducible as above. I can mount the drive on either machine. Machine 2 can `borg info` and Machine 1 cannot.

I have refrained from trying to create any more archives until I figure out why `borg info` won't run.

---

Thank you for any help you can provide!

**Comment by ThomasWaldmann** [comment]
On the machine that throws the traceback, the local chunks index or the related integrity information is corrupted / inaccessible.

Input/Output error is an error below borg, so either your hardware is malfunctioning or kernel / filesystem.

Do some hardware checks: RAM, hdd / ssd, see borg FAQ about IntegrityError.

After you've made sure your hardware is ok, you can try `borg delete --cache-only REPO` to clear and rebuild the local caches / indexes.

**Comment by acodispo** [comment]
@ThomasWaldmann Thanks for the amazingly quick response!

I've checked the hardware & everything seemed okay, so I tried `borg delete --cache-only REPO` as you suggested and that seems to have solved the problem. I've been able to create a new archive.

Would a corruption in the cache have been caused by a hardware issue? I'm trying to figure out what I did to cause this so I don't do it again. :)

Thanks again for your assistance!
